biorxiv/article_distances/05_biorxiv_article_distances_cosine.py

```diff
@@ -0,0 +1,440 @@
+)
+final_cosine_proxy_df.to_csv(
+    "output/annotated_links/article_distances_cosine_proxy.tsv", sep="\t", index=False
+)
+final_cosine_proxy_df.head()
+
+# ### Cosine Distance
+
+off_article_dist = cdist(
+    biorxiv_published.loc[PMC_published.index.tolist()]
+    .drop(["document", "preprint_doi", "published_doi", "pmcoa"], axis=1)
+    .values,
+    PMC_off_published.drop(["document", "journal"], axis=1).values,
+    "cosine",
+)
+off_article_dist.shape
+
+data = []
+for idx, row in tqdm.tqdm(articles_distance_cosine_df.iterrows()):
+    if row["journal"] in journal_mapper:
+        data.append(
+            {
+                "document": row["document"],
+                "pmcid": (
+                    PMC_off_published.query(f"journal=='{row['journal']}'")
+                    .reset_index()
+                    .document.values[0]
+                ),
+                "journal": row["journal"],
+                "distance": off_article_dist[idx, journal_mapper[row["journal"]]],
+            }
+        )
+
+final_cosine_df = articles_distance_cosine_df.assign(label="pre_vs_published").append(
+    pd.DataFrame.from_records(data).assign(label="pre_vs_random")
+)
+final_cosine_df.head()
+
+final_cosine_df = biorxiv_journal_df[["document", "preprint_doi"]].merge(
+    final_cosine_df
+)
+final_cosine_df.to_csv(
+    "output/annotated_links/article_distances_cosine.tsv", sep="\t", index=False
+)
+final_cosine_df.head()
+
+# # Distribution plot
+
+g = (
+    p9.ggplot(
+        final_original_df.replace(
+            {
+                "pre_vs_published": "preprint-published",
+                "pre_vs_random": "preprint-random",
+            }
+        )
+    )
+    + p9.aes(x="label", y="distance")
+    + p9.geom_violin(fill="#a6cee3")
+    + p9.labs(x="Document Pair Groups", y="Euclidean Distance")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=2)
+    + p9.theme(figure_size=(11, 8.5))
+)
+print(g)
+
+g = (
+    p9.ggplot(
+        final_cosine_proxy_df.replace(
+            {
+                "pre_vs_published": "preprint-published",
+                "pre_vs_random": "preprint-random",
+            }
+        )
+    )
+    + p9.aes(x="label", y="distance")
+    + p9.geom_violin(fill="#a6cee3")
+    + p9.labs(x="Document Pair Groups", y="Euclidean (L2 Norm) Distance")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=2)
+    + p9.theme(figure_size=(11, 8.5))
+)
+print(g)
+
+g = (
+    p9.ggplot(
+        final_cosine_df.replace(
+            {
+                "pre_vs_published": "preprint-published",
+                "pre_vs_random": "preprint-random",
+            }
+        )
+    )
+    + p9.aes(x="label", y="distance")
+    + p9.geom_violin(fill="#a6cee3")
+    + p9.labs(x="Document Pair Groups", y="Cosine Distance")
+    + p9.theme_seaborn(context="paper", style="ticks", font="Arial", font_scale=2)
+    + p9.theme(figure_size=(11, 8.5))
+)
+print(g)
```

Comment:
It's interesting how much closer to zero the cosine distance is for preprint-published pairs. I guess it makes sense that people would use it more